packages/unchained-client/src/evm/polygon/parser/__tests__/polygon.test.ts

```diff
@@ -0,0 +1,596 @@
+        blockTime: tx.timestamp,
+        address,
+        chainId: polygonChainId,
+        confirmations: tx.confirmations,
+        status: TxStatus.Confirmed,
+        fee: {
+          assetId: polygonAssetId,
+          value: '5618286173997000',
+        },
+        transfers: [
+          {
+            type: TransferType.Send,
+            to: address,
+            from: address,
+            assetId: polygonAssetId,
+            totalValue: '4079513530000000000',
+            components: [{ value: '4079513530000000000' }],
+          },
+          {
+            type: TransferType.Receive,
+            to: address,
+            from: address,
+            assetId: polygonAssetId,
+            totalValue: '4079513530000000000',
+            components: [{ value: '4079513530000000000' }],
+          },
+        ],
+      }
+
+      const actual = await txParser.parse(tx, address)
+
+      expect(expected).toEqual(actual)
+    })
+
+    it('should be able to parse token mempool', async () => {
+      const { txMempool } = tokenSelfSend
+      const address = '0xfA8a5D52aFCCAF40A0999ab9347D1Ba7Edc5395b'
+
+      const expected: ParsedTx = {
+        txid: txMempool.txid,
+        blockHeight: txMempool.blockHeight,
+        blockTime: txMempool.timestamp,
+        address,
+        chainId: polygonChainId,
+        confirmations: txMempool.confirmations,
+        status: TxStatus.Pending,
+        transfers: [],
+      }
+
+      const actual = await txParser.parse(txMempool, address)
+
+      expect(expected).toEqual(actual)
+    })
+
+    it('should be able to parse token', async () => {
+      const { tx } = tokenSelfSend
+      const address = '0xfA8a5D52aFCCAF40A0999ab9347D1Ba7Edc5395b'
+
+      const expected: ParsedTx = {
+        txid: tx.txid,
+        blockHash: tx.blockHash,
+        blockHeight: tx.blockHeight,
+        blockTime: tx.timestamp,
+        address,
+        chainId: polygonChainId,
+        confirmations: tx.confirmations,
+        status: TxStatus.Confirmed,
+        fee: {
+          assetId: polygonAssetId,
+          value: '12798989060278680',
+        },
+        transfers: [
+          {
+            type: TransferType.Send,
+            from: address,
+            to: address,
+            assetId: 'eip155:137/erc20:0x2791bca1f2de4661ed88a30c99a7a9449aa84174',
+            totalValue: '700000000',
+            components: [{ value: '700000000' }],
+            token: usdcToken,
+          },
+          {
+            type: TransferType.Receive,
+            from: address,
+            to: address,
+            assetId: 'eip155:137/erc20:0x2791bca1f2de4661ed88a30c99a7a9449aa84174',
+            totalValue: '700000000',
+            components: [{ value: '700000000' }],
+            token: usdcToken,
+          },
+        ],
+      }
+
+      const actual = await txParser.parse(tx, address)
+      expect(expected).toEqual(actual)
+    })
+  })
+
+  describe('erc20', () => {
```

Comment:
Ditto, do we want to use MRC20 terminology?

Reply:
Don't think so, MRC20 is not a thing unlike ex. BEP20

Reply:
Yeah seems like matic is a MRC20 akin to avax being an arc20, but generally speaking the tokens are various forms of wrapped erc20 contracts. Think it's best to leave as is